synapse/module_api/__init__.py

```diff
@@ -89,7 +89,7 @@ def check_user_exists(self, user_id):
         return defer.ensureDeferred(self._auth_handler.check_user_exists(user_id))
 
     @defer.inlineCallbacks
-    def register(self, localpart, displayname=None, emails=[]):
+    def register(self, localpart, displayname=None):
```

Comment:
We should probably note in the changelog that his changed (or leave it for backwards compatibility).

Do we know if any external modules called this with an iterable of emails?

Reply:
Aha! Good spot. Ok, I was under the impression that modules like matrix-synapse-ldap3 bound emails in another manner (through the `check_3pid_auth` module method).

But [you're right](https://github.com/matrix-org/matrix-synapse-ldap3/blob/490b2bf844b3dab2933fb92029714fbeb9e086b0/ldap_auth_provider.py#L294-L296)!

Which probably makes this whole PR defunct I suppose.

Reply:
Ah, yeah looks like this is still used then. :)

Reply:
Oh well!